storage/innobase/include/trx0trx.h

```diff
@@ -734,14 +734,18 @@ struct trx_t {
 	Transitions to COMMITTED are protected by trx_t::mutex. */
 	trx_state_t	state;
 #ifdef WITH_WSREP
-	/** whether wsrep_on(mysql_thd) held at the start of transaction */
-	bool		wsrep;
+	/** (1) whether wsrep_on(mysql_thd) held at the start of transaction
+	    (2) whether wsrep_thd_is_BF(mysql_thd) held at start of transaction
+	    implies (1) */
+	byte		wsrep;
 	bool is_wsrep() const { return UNIV_UNLIKELY(wsrep); }
+	bool is_wsrep_BF() const { return UNIV_UNLIKELY(wsrep == 2); }
```

Comment:
This may conflict with what is already in 10.6 https://github.com/codership/mariadb-server/blob/10.6/storage/innobase/include/trx0trx.h#L719